pkg/p2p/p2p.go

```diff
@@ -59,6 +59,7 @@ type StreamerDisconnecter interface {
 type Stream interface {
 	io.ReadWriter
 	io.Closer
+	ResponseHeaders() Headers
```

Comment:
I find this addition to the interface a bit contentious. I think it would be cleaner to just return the response headers together with the new stream on `NewStream`. I.e. change the method signature for `NewStream` to be:
`NewStream(ctx context.Context, overlay swarm.Address, headers p2p.Headers, protocolName, protocolVersion, streamName string) (p2p.Stream, p2p.Headers, error)`. I'd like to hear what other think. cc @zelig @janos

Reply:
With the suggestion to return headers, I see the problem to clearly identify that response headers are returned. With the interface method, it is more obvious, so I would go with the method.